python/ray/experimental/state.py

```diff
@@ -304,3 +303,20 @@ def client_table(self):
       node_info[node_ip_address].append(client_info_parsed)
 
     return node_info
+
+  def task_profiles(self):
+    """Fetch and return a list of task profiles.
+
+    Returns:
+      A list of task profiles.
+    """
+    r = ray.worker.global_worker.redis_client
+    event_names = r.keys("event_log*")
+    results = []
+    for i in range(len(event_names)):
+      event_list = r.lrange(event_names[i], 0, -1)
+      decoded = []
+      for event in event_list:
+        decoded.append(event.decode("ascii"))
```

Comment:
in this case, I think you also want to run `json.loads` on `event.decode("ascii")`

and you'll also have to do `import json` at the top of the file